### PR TITLE
Deflake e2e tests

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind failing-test

#### Which issue(s) this PR fixes:

None

#### Does this PR have test?

None

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
For the vagrant based test its DNS:
```
curl: (6) Could not resolve host: metrics.security-profiles-operator
```

**Comment by saschagrunert** [comment]
@JAORMX @jhrozek PTAL

File: hack/ci/Vagrantfile [hunk 1]
```diff
@@ -31,6 +30,16 @@ Vagrant.configure("2") do |config|
...
+
+      echo "nameserver 8.8.8.8" > /etc/resolv.conf
+      modprobe br_netfilter
+      ip6tables --list >/dev/null
+      iptables -F
+      sysctl -w net.ipv4.conf.all.route_localnet=1
+      sysctl -w net.ipv4.ip_forward=1
+      sysctl -w net.bridge.bridge-nf-call-iptables=1
+      sysctl -w fs.inotify.max_user_watches=1048576
+      iptables -t nat -I POSTROUTING -s 127.0.0.0/8 ! -d 127.0.0.0/8 -j MASQUERADE
```
Comment (JAORMX) [inline hack/ci/Vagrantfile:42]: Mind adding a comment or a description on the commit message of why all of this is needed?

Comment (JAORMX) [reply hack/ci/Vagrantfile:42]: uhm... why do we need that postrouting iptables rules?

File: hack/ci/env.sh [hunk 2]
```diff
@@ -27,3 +27,5 @@ IP=$(local_ip)
 export KUBE_MASTER_URL=$IP
 export KUBE_MASTER_IP=$IP
 export KUBE_MASTER=$IP
+
+alias k=kubectl
```
Comment (JAORMX) [inline hack/ci/env.sh:31]: This seems unused, or was this added for debugging purposes (when you logged into the VM). If so, let's add a comment about it.

File: test/suite_test.go [hunk 3]
```diff
@@ -247,6 +247,7 @@ func (e *kinde2e) TearDownTest() {
 	e.logf("#### Snapshot of security-profiles-operator namespace ####")
 	e.kubectl("--namespace", "security-profiles-operator", "describe", "all")
 	e.logf("########")
+	e.kubectl("-n", "kube-system", "logs", "deploy/coredns")
```
Comment (JAORMX) [inline test/suite_test.go:250]: Why do we need the coredns logs?

**Review by JAORMX — APPROVED** [review]

**Comment by saschagrunert** [comment]
Following up in https://github.com/kubernetes-sigs/security-profiles-operator/pull/482

**Pull request merged**